attestation-service/src/verifier/cca/mod.rs

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) 2023 Arm Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+extern crate serde;
+extern crate veraison_apiclient;
+
+use super::*;
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+use core::result::Result::Ok;
+use serde::{Deserialize, Serialize};
+use serde_json::{Map, Value};
+use veraison_apiclient::*;
+
+const VERIFIER_ADDR: &str = "VERIFIER_ADDR";
+const DEFAULT_VERIFIER_ADDR: &str = "localhost:8080";
+
+#[derive(Debug, Default)]
+pub struct CCA {}
+
+#[derive(Serialize, Deserialize)]
+struct CcaEvidence {
+    // CCA token
+    token: Vec<u8>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct RealmToken {
+    //cca_realm_personalization_value: String,
+    cca_realm_initial_measurement: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct Evidence {
+    cca_realm_delegated_token: RealmToken,
+}
+
+fn my_evidence_builder(
+    nonce: &[u8],
+    accept: &[String],
+    token: Vec<u8>,
+) -> Result<(Vec<u8>, String), veraison_apiclient::Error> {
+    log::info!("server challenge: {:?}", nonce);
+    log::info!("acceptable media types: {:#?}", accept);
+    // TODO: Get the CCA media type from the slice of `accept`.
+    Ok((
+        token,
+        "application/eat-collection; profile=http://arm.com/CCA-SSD/1.0.0".to_string(),
+    ))
+}
+
+#[allow(unused_variables)]
```

Comment:
not using the `nonce` but instead leave the `rust-apiclient` to create a new nonce.

Reply:
Well. If the `nonce` parameter is not used, you can change the variable name to `_nonce` with a `_` prefix to pass lint error. 

Besides, the `nonce` here should be checked with the one wrapped inside the CCA evidence (I remembered that the name is `challenge`), which is very important in KBS protocol to prevent the reply attack from the client (attestation-agent) side. Please refer to other plugins like tdx and snp's.

Reply:
I see your point!

the `nonce` here (L82) is used to build a session which will talk with the Veraison, it is serving as a different purpose with the `challenge` from CCA.

To compare the `nonce` and `challenge(10)` is possible after the token is parsed and build a JSON representative of CCA token, Can we leave this as a follow up? I will come back to fix it up as there is still some pending discussion around the token parse.